comms/src/protocol/messaging/protocol.rs

```diff
@@ -205,7 +206,9 @@ impl MessagingProtocol {
 
     pub fn framed<TSubstream>(socket: TSubstream) -> Framed<IoCompat<TSubstream>, LengthDelimitedCodec>
     where TSubstream: AsyncRead + AsyncWrite + Unpin {
-        Framed::new(IoCompat::new(socket), LengthDelimitedCodec::new())
+        let mut codec = LengthDelimitedCodec::new();
+        codec.set_max_frame_length(MAX_FRAME_LENGTH_MB * 1_024 * 1_024);
+        Framed::new(IoCompat::new(socket), codec)
```

Comment:
nit: The builder is the idiomatic way to customize the codec

```suggestion
        let codec = LengthDelimitedCodec::builder()
        .max_frame_length(MAX_FRAME_LENGTH)
        .new_codec();
        Framed::new(IoCompat::new(socket), codec)
```

Reply:
Committed, rebased and force pushed from local